homeassistant/components/ihc/sensor.py

```diff
@@ -51,29 +51,12 @@ def __init__(
     ) -> None:
         """Initialize the IHC sensor."""
         super().__init__(ihc_controller, controller_id, name, ihc_id, product)
-        self._state = None
-        self._unit_of_measurement = unit
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return (
-            SensorDeviceClass.TEMPERATURE
-            if self._unit_of_measurement in TEMPERATURE_UNITS
-            else None
+        self._attr_native_unit_of_measurement = unit
+        self._attr_device_class = (
+            SensorDeviceClass.TEMPERATURE if unit in TEMPERATURE_UNITS else None
         )
```

Comment:
This is a little cleaner (`None` is already the default)
```suggestion
        if unit in TEMPERATURE_UNITS:
            self._attr_device_class = SensorDeviceClass.TEMPERATURE
```